packages/preferences/src/node/json-preference-server.ts

```diff
@@ -100,27 +101,39 @@ export class JsonPreferenceServer implements PreferenceServer {
             if (this.preferences) {
```

Comment:
I think the only change that which class requires is to send one event at the end of this method instead of multiple.